### ISSUE TITLE
A question about the x-ms-version bump

### ISSUE THREAD

**Issue by Stephan202** [Azure/azure-storage-java#156]
Preface: I'm not very familiar with the Azure Storage Services; my experience is limited to maintaining a small piece of integration logic which interfaces with the Storage account of a third party. I apologize in advance if this issue tracker is not the right forum for this type of question.

Issue:
* We upgraded from `azure-storage-java` 5.0.0 to 5.1.0. Code compiled just fine. :white_check_mark: 
* Using a test account we performed a blob upload. This went fine as well. :white_check_mark: 
* When deployed to production the upload to the third-party's storage account failed. :x:

After enabling debug logging, we found the culprit:
```
15:16:32.281 [main] DEBUG ROOT - {1c9c6b00-f759-43ab-a77c-f25bf30fc3f4}: {Error response received. HttpStatusCode= 400, HttpStatusMessage= The value for one of the HTTP headers is not in the correct format., ErrorCode= InvalidHeaderValue, ExtendedErrorInformation= {ErrorMessage= The value for one of the HTTP headers is not in the correct format..RequestId:930e249c-0001-0139-0fb4-cc6d64000000.Time:2017-05-14T13:16:30.0939372Z, AdditionalDetails= { HeaderValue= 2016-10-16,HeaderName= x-ms-version}}}
```

I understand that the `x-ms-version` header is used for API versioning. And I see that for `azure-storage-java` version 5.1.0 the targeted storage API was [bumped](https://github.com/Azure/azure-storage-java/compare/v5.0.0...v5.1.0#diff-7d32b8641d0a321f0b3ef2ba9d80175dR644) from `2016-05-31` to `2016-10-16`. But the [documentation](https://docs.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services) indicates that version `2016-05-31` is still the latest.

So my question is: is this version not officially released yet (i.e., rolled out to some environments/accounts but not to others), or should I reach out to the owner of the production account and ask them to accept this new version?

NB: Is the 5.0.0 -> 5.1.0 version change supposed to be a [semantic version change](http://semver.org/)? As a novice user I would have expected this upgrade to be backwards compatible.

**Comment by jofriedm-msft** [comment]
Hi @Stephan202 , thank you for pointing this out. The change in version was not meant to be part of the 5.1.0 release. As a short term fix a 5.1.1 release has been issued both on github and maven.

**Comment by Stephan202** [comment]
@jofriedm-msft, thanks for the rapid response. I can confirm that the simple blob upload test we have here now passes against both environments. Will close the ticket accordingly :).
